### ISSUE TITLE
cy.origin re-redirect

### ISSUE THREAD

**Issue by stychu** [cypress-io/cypress#23236]
### Current behavior

My login page is on different domain than the app.
I have a redirect within the app that redirects to the login domain.
Trying to make it work with `cy.origin` but it seems it doesn't pick up middle redirect from the app or I'm doing something wrong.

Flow
1. Visit http://xyz/baseurl/create-location (not logged in before)
2. Auto redirect to http://xyz/baseurl/one-login-redirect
3. http://xyz/baseurl/one-login-redirect redirects to http://abc/login

This doesn't work because of the flow step 2 which triggers second redirect from within the app then goes to the login domain. I can see in the cypress my login domain loaded but cy.origin can't query the dom for the inputs to login

Seems like cy.origin expects direct redirect trigger to properly work.
Can i somehow overcome this? Tried to assert he step 2 url but `cy.origin` still doesn't work 

### Desired behavior

`cy.origin` should work when redirect triggers other redirects (middle redirects) leading to origin

### Test code to reproduce

```
    cy.visit('/create-location')
    cy.origin(Cypress.env('LOGIN_ORIGIN'), { args: users.ADMIN }, ({ username, password }) => {
      cy.get('#username').type(username)
      cy.get('#password').type(password)
      cy.get('button[type=submit]').click()
    })
```

### Cypress Version

10.4.0

### Other

<img width="434" alt="Screenshot 2022-08-10 at 11 57 53" src="https://user-images.githubusercontent.com/9383657/183874905-78a6b182-2c4c-4820-9657-c3e6d8b101f1.png">

**Comment by AtofStryker** [comment]
Hey @stychu. Thank you for opening an issue (hopefully this one is simpler 😅 ). Just have a few questions:

1. Is your main application and login application on different origins? They both look to be `localhost`, which might mean you may not need `cy.origin`
2. Is your base URL set?
3. Anyway you could throw together a quick reproduction repo?

This looks like a `302` server side redirect. We have a similar issue opened with #22282 where this happens with a `301` redirect. Does the origin command time out or does the get fail? Are you able to provide the Assertion error?

**Comment by stychu** [comment]
@AtofStryker 
1. Yes. Internally we use OneLogin flow. Locally we use WireMock to mock the OneLogin login form submission
2. Yes
3. Im bit busy rn but I will try 

> This looks like a 302 server side redirect. We have a similar issue opened with https://github.com/cypress-io/cypress/issues/22282 where this happens with a 301 redirect. Does the origin command time out or does the get fail? Are you able to provide the Assertion error?

Indeed
<img width="308" alt="Screenshot 2022-08-11 at 12 24 39" src="https://user-images.githubusercontent.com/9383657/184117710-51d47688-3d79-41a6-a991-4cd568e25692.png">

As described in the #22282 if I would trigger the OneLogin origin via button click for example from my app at /login it would do direct redirect to OneLogin origin and cy.origin works like a charm. Problem is I want to test a scenario where user gets redirected to login if not authenticated but he tries to access any page of the app. In this case server redirect doesnt seem to work with origin

**Comment by AtofStryker** [comment]
We confirmed that #23297 should fix #22282. We can't 💯 confirm since we do not have a concrete reproduction, but we feel confident that #23297 should resolve this issue.

**Comment by stychu** [comment]
> We confirmed that #23297 should fix #22282. We can't 💯 confirm since we do not have a concrete reproduction, but we feel confident that #23297 should resolve this issue.

Awesome! As soon as it lands I will try it out. 
Sorry for not providing a reproduction but I tried to make it up but failed. I hope for the fixes
